Library/Homebrew/api/formula.rb

```diff
@@ -33,7 +33,7 @@ def all_formulae
             if cached_formula_json_file.exist?
               last_modified = cached_formula_json_file.mtime.utc
               last_modified = last_modified.strftime("%a, %d %b %Y %H:%M:%S GMT")
-              curl_args = ["--header", "If-Modified-Since: #{last_modified}", *curl_args]
+              curl_args = ["--time-cond", last_modified, *curl_args] unless cached_formula_json_file.empty?
```

Comment:
It seems like we can pass `cached_formula_json_file` to `--time-cond` to avoid having to get its `mtime`. Do we also want to use `--remote-time` here?

Reply:
`curl_download` adds `--remote-time` automatically

Reply:
So I guess this whole `if` block can be something like
```ruby
if cached_formula_json_file.exist? && !cached_formula_json_file.empty?
  curl_args.unshift("--time-cond", cached_formula_json_file)
end
```
or even as a single line, if it fits the line length limit.

Reply:
Yep. Might need a `#to_s` on the `cached_formula_json_file`, I'm not sure off the top of my head.

Reply:
This modified `if` block works, though as a one-liner it's too long for `brew style` to allow. I just pushed this change to my branch.